compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -40,7 +40,8 @@ object Inliner {
   private val ContextualImplicit = new Property.StickyKey[Unit]
 
   def markContextualImplicit(tree: Tree)(implicit ctx: Context): Unit =
-    methPart(tree).putAttachment(ContextualImplicit, ())
+    if (!defn.ScalaPredefModule.moduleClass.derivesFrom(tree.symbol.maybeOwner))
```

Comment:
Is this only for performance?

Reply:
No, a Predef import might also shadow a real at the call site. Since there are lots of implicits this is hard to predict. I had a case where a Predef$conforms shadowed a better binding (it was in one of the examples that Miles came up with). Predef is effectively global, since it is
imported by default into every module. So it makes sense to treat it as global.